subsys/bluetooth/controller/ll_sw/ull_filter.c

```diff
@@ -1014,8 +1014,12 @@ static int rl_access_check(bool check_ar)
 		}
 	}
 
+	/* NOTE: Allowed when passive scanning, otherwise deny if advertising,
+	 *       active scanning, initiating or periodic sync create is active.
+	 */
 	return ((IS_ENABLED(CONFIG_BT_BROADCASTER) && ull_adv_is_enabled(0)) ||
-		(IS_ENABLED(CONFIG_BT_OBSERVER) && ull_scan_is_enabled(0)))
+		(IS_ENABLED(CONFIG_BT_OBSERVER) &&
+		 (ull_scan_is_enabled(0) & ~ULL_SCAN_IS_PASSIVE)))
 		? 0 : 1;
```

Comment:
While this looks correct, the double negation makes it hard to read imo. 

Consider doing something like

```
	if (IS_ENABLED(CONFIG_BT_BROADCASTER) && ull_adv_is_enabled(0)) {
		return 0;
	} else if (IS_ENABLED(CONFIG_BT_OBSERVER) &&
		   (ull_scan_is_enabled(0) & ULL_SCAN_IS_PASSIVE) == 0) {
		return 0;
	}

	return 1;
```

Which not only splits the broadcaster check from the observer check, but also makes the check for passive much clearer in my opinion.